util/reflect.go

```diff
@@ -23,26 +23,42 @@ import (
 
 // IsTypeStructPtr reports whether v is a struct ptr type.
 func IsTypeStructPtr(t reflect.Type) bool {
+	if t == reflect.TypeOf(nil) {
```

Comment:
This is a bit messy - but I don't see a different way to do this. It's a shame `reflect.Type` doesn't have `IsNil` or `IsValid` or somesuch.

Reply:
Another annoyance in a long list around reflect :-)